plugins/lua/sort/places.lua

```diff
@@ -77,6 +85,31 @@ local function get_zone_search_key(zone)
     return table.concat(result, ' ')
 end
 
+local function get_location_search_key(zone)
+    local site = df.global.world.world_data.active_site[0]
+    local result = {}
+
+    -- get language_name and type (we dont need user-given zone name because it does not appear on this page)
+    local building, success, _ = utils.binsearch(site.buildings, zone.location_id, 'id')
+    if success and building.name then
+        table.insert(result, language_name_types[building.name.type] or '')
+        if building.name.has_name then
+            table.insert(result, dfhack.TranslateName(building.name, true))
+        end
+
+        -- for temples and guildhalls, get assigned organization
+        if df.abstract_building_templest:is_instance(building) then
+            table.insert(result, get_location_religion(building.deity_data.Deity or building.deity_data.Religion, building.deity_type))
+        elseif df.abstract_building_guildhallst:is_instance(building) then
+            local dwarfified_profession = locationselector.get_profession_string(building.contents.profession):gsub('[Mm][Aa][Nn]', 'dwarf') -- Craftsman becomes Craftsdwarf, etc
```

Comment:
"dwarf"ifying is a good idea, but I'd suggest adding instead of replacing (in case people do actually search for "craftsman"). so the string would become `CRAFTSMAN CRAFTSdwarf` (capitalization doesn't matter). I'd also suggest moving this logic to `locationselector.get_profession_string` so the location selector screens can benefit from it too.